### PR TITLE
Improve time accounting for Incoming

### PR DESCRIPTION
**Pull request by Ralith**
Plus some bonus MSRV cleanup.

### CODE REVIEW

File: quinn-udp/src/windows.rs [hunk 1]
```diff
@@ -403,7 +402,7 @@ const CMSG_LEN: usize = 128;
 const OPTION_ON: u32 = 1;
 
 // FIXME this could use [`std::sync::OnceLock`] once the MSRV is bumped to 1.70 and upper
```
Comment (mxinden) [inline quinn-udp/src/windows.rs:404]:
```suggestion
```

**Review by mxinden — APPROVED** [review]
+1 on the `quinn-udp/` changes. Thanks.

**Review by djc — APPROVED** [review]
Nice!

**Comment by djc** [comment]
LazyLock is documented as stable from 1.80, but we have an older MSRV, right? Why isn't CI flagging this?

**Comment by Ralith** [comment]
Good question. Dropping the MSRV stuff for now.

**Review by djc — APPROVED** [review]

**Pull request merged**